collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh

```diff
@@ -526,6 +536,70 @@ rke2-k8s() {
 
 }
 
+kubeadm-k8s() {
+  
+  KUBEADM_DIR="/etc/kubernetes/"
+  KUBEADM_STATIC_DIR="/etc/kubernetes/manifests/"
+  techo "Collecting k8s kubeadm cluster logs"
+  if [ -f /$USER/.kube/config ]; then
+    mkdir -p $TMPDIR/kubeadm/kubectl
+    KUBECONFIG=/$USER/.kube/config
+    /usr/bin/kubectl --kubeconfig=$KUBECONFIG get nodes -o wide > $TMPDIR/kubeadm/kubectl/nodes 2>&1
+    /usr/bin/kubectl --kubeconfig=$KUBECONFIG describe nodes > $TMPDIR/kubeadm/kubectl/nodesdescribe 2>&1
+    /usr/bin/kubectl --kubeconfig=$KUBECONFIG version > $TMPDIR/kubeadm/kubectl/version 2>&1
+    /usr/bin/kubectl --kubeconfig=$KUBECONFIG get pods -o wide --all-namespaces > $TMPDIR/kubeadm/kubectl/pods 2>&1
+    /usr/bin/kubectl --kubeconfig=$KUBECONFIG get svc -o wide --all-namespaces > $TMPDIR/kubeadm/kubectl/services 2>&1
+    /usr/bin/kubectl --kubeconfig=$KUBECONFIG cluster-info dump > $TMPDIR/kubeadm/kubectl/cluster-info_dump 2>&1
+  fi
+
+  if [ -f /$USER/.kube/config ]; then
+    KUBECONFIG=/$USER/.kube/config
+    /usr/bin/kubectl --kubeconfig=$KUBECONFIG api-resources > $TMPDIR/kubeadm/kubectl/api-resources 2>&1
+    KUBEADM_OBJECTS=(clusterroles clusterrolebindings crds mutatingwebhookconfigurations namespaces nodes pv validatingwebhookconfigurations)
+    KUBEADM_OBJECTS_NAMESPACED=(apiservices configmaps cronjobs deployments daemonsets endpoints events helmcharts hpa ingress jobs leases pods pvc replicasets roles rolebindings statefulsets)
+    for OBJECT in "${KUBEADM_OBJECTS[@]}"; do
+      /usr/bin/kubectl --kubeconfig=$KUBECONFIG get ${OBJECT} -o wide > $TMPDIR/kubeadm/kubectl/${OBJECT} 2>&1
+    done
+    for OBJECT in "${KUBEADM_OBJECTS_NAMESPACED[@]}"; do
+      /usr/bin/kubectl --kubeconfig=$KUBECONFIG get ${OBJECT} --all-namespaces -o wide > $TMPDIR/kubeadm/kubectl/${OBJECT} 2>&1
+    done
+  fi
+
+  mkdir -p $TMPDIR/kubeadm/podlogs
+  techo "Collecting k8s kubeadm system pod logs"
+  if [ -f /$USER/.kube/config ]; then
```

Comment:
I think these if checks can be done once at the top of the function, not needed for every section.